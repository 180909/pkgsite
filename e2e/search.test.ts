/**
 * @license
 * Copyright 2021 The Go Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style
 * license that can be found in the LICENSE file.
 */

import './global-types';
import puppeteer, { Page } from 'puppeteer';

describe('Search', () => {
  let page: Page;

  beforeAll(async () => {
    page = await newPage();
    await page.goto(baseURL + '/search?q=http');
  });

  afterAll(async () => {
    await page.close();
  });

  test('accessibility tree matches snapshot', async () => {
    await prepare(page);
    const a11yTree = await page.accessibility.snapshot();
    expect(a11yTree).toMatchSnapshot();
  });

  test('desktop viewport matches image snapshot', async () => {
    await prepare(page);
    const image = await page.screenshot({ fullPage: true });
    expect(image).toMatchImageSnapshot();
  });

  test('mobile viewport matches image snapshot', async () => {
    await page.emulate(puppeteer.devices['Pixel 2']);
    await prepare(page);
    const image = await page.screenshot({ fullPage: true });
    expect(image).toMatchImageSnapshot();
  });
});

/**
 * prepare gets the page ready for snapshot tests by rewriting highly
 * variable page content to constant values.
 * @param page The page to prepare
 */
async function prepare(page: Page): Promise<void> {
  await Promise.all([
    page.$$eval('[data-test-id="snippet-title"]', els =>
      els.map(el => {
        el.innerHTML = 'net/http/pprof';
        (el as HTMLAnchorElement).href = 'net/http/pprof';
      })
    ),
    page.$$eval('[data-test-id="snippet-synopsis"]', els =>
      els.map(el => {
        el.innerHTML =
          'Package pprof serves via its HTTP server runtime profiling ' +
          'data in the format expected by the pprof visualization tool.';
      })
    ),
    page.$$eval('[data-test-id="snippet-version"]', els =>
      els.map(el => (el.innerHTML = 'go1.16.3'))
    ),
    page.$$eval('[data-test-id="snippet-published"]', els =>
      els.map(el => (el.innerHTML = 'Apr 1, 2021'))
    ),
    page.$$eval('[data-test-id="snippet-importedby"]', els =>
      els.map(el => (el.innerHTML = '11632'))
    ),
    page.$$eval('[data-test-id="snippet-license"]', els =>
      els.map(el => (el.innerHTML = 'BSD-3-Clause'))
    ),
  ]);
}
