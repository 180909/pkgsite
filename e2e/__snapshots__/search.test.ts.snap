// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility tree (desktop) 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Black Lives Matter",
      "role": "text",
    },
    Object {
      "name": "Support the Equal Justice Initiative",
      "role": "link",
    },
    Object {
      "name": "Link to Go homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
      "value": "http",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "name": "About",
      "role": "link",
    },
    Object {
      "level": 1,
      "name": "Results for “http”",
      "role": "heading",
    },
    Object {
      "name": "Search help",
      "role": "link",
    },
    Object {
      "name": "1 – 10 of 100 results",
      "role": "text",
    },
    Object {
      "name": "Previous",
      "role": "text",
    },
    Object {
      "name": "1",
      "role": "text",
    },
    Object {
      "name": "2",
      "role": "link",
    },
    Object {
      "name": "3",
      "role": "link",
    },
    Object {
      "name": "4",
      "role": "link",
    },
    Object {
      "name": "5",
      "role": "link",
    },
    Object {
      "name": "Next",
      "role": "link",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "Previous",
      "role": "text",
    },
    Object {
      "name": "1",
      "role": "text",
    },
    Object {
      "name": "2",
      "role": "link",
    },
    Object {
      "name": "3",
      "role": "link",
    },
    Object {
      "name": "4",
      "role": "link",
    },
    Object {
      "name": "5",
      "role": "link",
    },
    Object {
      "name": "Next",
      "role": "link",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "http - Search Results · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`accessibility tree (mobile) 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Black Lives Matter",
      "role": "text",
    },
    Object {
      "name": "Support the Equal Justice Initiative",
      "role": "link",
    },
    Object {
      "name": "Link to Go homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
      "value": "http",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "name": "About",
      "role": "link",
    },
    Object {
      "level": 1,
      "name": "Results for “http”",
      "role": "heading",
    },
    Object {
      "name": "Search help",
      "role": "link",
    },
    Object {
      "name": "1 – 10 of 100 results",
      "role": "text",
    },
    Object {
      "name": "Previous",
      "role": "text",
    },
    Object {
      "name": "1",
      "role": "text",
    },
    Object {
      "name": "2",
      "role": "link",
    },
    Object {
      "name": "3",
      "role": "link",
    },
    Object {
      "name": "4",
      "role": "link",
    },
    Object {
      "name": "5",
      "role": "link",
    },
    Object {
      "name": "Next",
      "role": "link",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "net/http/pprof",
      "role": "link",
    },
    Object {
      "name": "Package pprof serves via its HTTP server runtime profiling data in the format expected by the pprof visualization tool.",
      "role": "text",
    },
    Object {
      "name": "Version:",
      "role": "text",
    },
    Object {
      "name": "go1.16.3",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Published:",
      "role": "text",
    },
    Object {
      "name": "Apr 1, 2021",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "Imported by:",
      "role": "text",
    },
    Object {
      "name": "11632",
      "role": "text",
    },
    Object {
      "name": "|",
      "role": "text",
    },
    Object {
      "name": "License:",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "text",
    },
    Object {
      "name": "Previous",
      "role": "text",
    },
    Object {
      "name": "1",
      "role": "text",
    },
    Object {
      "name": "2",
      "role": "link",
    },
    Object {
      "name": "3",
      "role": "link",
    },
    Object {
      "name": "4",
      "role": "link",
    },
    Object {
      "name": "5",
      "role": "link",
    },
    Object {
      "name": "Next",
      "role": "link",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "http - Search Results · pkg.go.dev",
  "role": "WebArea",
}
`;
